### ISSUE TITLE
verify-owners: reading missing files

### ISSUE THREAD

**Issue by stevekuznetsov** [kubernetes/test-infra#10541]
**What happened**:
```json
{
  "author": "leseb",
  "component": "hook",
  "error": "open /tmp/git423918455/playbooks/openshift-prometheus/OWNERS: no such file or directory",
  "event-GUID": "6e40cc90-054a-11e9-961a-7dfbe2ee70dd",
  "event-type": "pull_request",
  "level": "error",
  "msg": "Failed to read /tmp/git423918455/playbooks/openshift-prometheus/OWNERS.",
  "org": "openshift",
  "plugin": "verify-owners",
  "pr": 10914,
  "repo": "openshift-ansible",
  "time": "2018-12-21T18:01:30Z",
  "url": "https://github.com/openshift/openshift-ansible/pull/10914"
}
```

**What you expected to happen**:
Not try to read a file that does not exist. Are we reading incorrect branches?

/kind bug

/cc @cjwagner @matthyx

**Comment by matthyx** [comment]
Will try to look at it tomorrow. Thx!

**Comment by cjwagner** [comment]
Hmm, we should have the right git state when we are doing the validation because we checkout the PR here: https://github.com/kubernetes/test-infra/blob/d86158482b48d65327715939125b31f4395eb136/prow/plugins/verify-owners/verify-owners.go#L110
We only loop over modified files and that file doesn't seem to be touched by the PR so I'm guessing that the author force pushed to the PR branch and there was a race with checking out the PR since we don't specify a commit SHA when we checkout the PR.

**Comment by cjwagner** [comment]
The base branch also changed multiple times. That could be related: 
https://github.com/openshift/openshift-ansible/pull/10914#event-2038903967
https://github.com/openshift/openshift-ansible/pull/10914#event-2041353252

The later coincides with the timestamp on the error message.

**Comment by matthyx** [comment]
Should we then specify a commit SHA to avoid this race?

**Comment by stevekuznetsov** [comment]
Very interesting -- in the end this does seem like it has to be a race between changes to the PR and us resolving the ref, right? Maybe we should just downgrade this from an error.

**Comment by cjwagner** [comment]
The race is definitely a problem, but the changing base branch could be what is causing the error here. That seems possible to me since we force push all the time and haven't seen this error, but we rarely change the base branch. Did you see any other similar errors?

If we're going to address this we may as well have it set the PR branch to the commit we expect since we know the commit SHA that we want. Downgrading this from an error is just hiding the actual problem.
That being said we still don't have good protection against races because we handle webhook events concurrently so I'm not sure this is worth trying to actually solve as we won't be able to avoid this type of race anyways. If thats the case we should just downgrade the error like you are suggesting since they are rare and the failure mode is safe.

**Comment by matthyx** [comment]
@stevekuznetsov I think that one was solved by #10613 ?
